dist/bin/dotr

```diff
@@ -31,18 +31,39 @@ source "$PROG_HOME/bin/common"
 
 CLASS_PATH=".$PSEP$DOTTY_LIB$PSEP$SCALA_LIB"
 
-# -d must be the first option
-case "$1" in
-  -d) DEBUG="$DEBUG_STR" && shift ;;
-esac
 
-first_arg="$1"
+addResidual () {
+  residual_args+=("$1")
+}
 
-if [ -z "$1" ]; then
+run_repl=true
+run_app=flase
+while [[ $# -gt 0 ]]; do
+key=$1
+case $key in
+        -repl)
+         run_repl=true
+         shift
+         ;;
+        -classpath)
+         CLASS_PATH+=":$2"
+          shift
+          shift
+          ;;
+          *)
+           addResidual "$1"
+           shift
+           ;;
+        -d)
+        DEBUG="$DEBUG_STR"
+         shift
+         ;;
+  esac
+done
+
+if [ -z $residual_args ]; then
     echo "Starting dotty REPL..."
     eval "$PROG_HOME/bin/dotc -repl"
-elif [[ ${first_arg:0:1} == "-" ]]; then
-    eval "$PROG_HOME/bin/dotc -repl $@"
```

Comment:
We should still be able to call the repl with a custom classpath and custom args. You can assume that the `-repl` is set in that case and hence `run_repl` should be true.